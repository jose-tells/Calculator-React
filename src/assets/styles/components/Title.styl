.header
  display flex
  justify-content center
  margin-bottom 16px
  .header__title
    text-align center
    font-size titlesFontSize
    font-weight 800
    color white