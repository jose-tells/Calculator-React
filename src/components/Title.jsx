import React from 'react';

const Title = () => {
  return (
    <header className='header'>
      <h1 className='header__title'>
        Hola!
        <br />
        Â¿Que quieres
        <br />
        calcular?
      </h1>
    </header>
  );
};

export default Title;
