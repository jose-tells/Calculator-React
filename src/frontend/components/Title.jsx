import React from 'react';

const Title = () => {
  return (
    <header className='header'>
      <h1 className='header__title'>
        Hi!
        <br />
        What do you want to
        <br />
        calculate?
      </h1>
    </header>
  );
};

export default Title;
