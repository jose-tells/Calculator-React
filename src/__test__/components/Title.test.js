import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import Title from '../../components/Title';

describe('<Title />', () => {
  const title = mount(<Title />);
  test('Title component render', () => {
    expect(title.length).toEqual(1);
  });
  test('Title text', () => {
    expect(title.find('.header__title').text()).toEqual('Hi!What do you want tocalculate?');
  });

  test('Title h1 tags', () => {
    expect(title.find('h1')).toHaveLength(1);
  });

  test('Title <br> tags', () => {
    expect(title.find('br')).toHaveLength(2);
  });

  test('Title component UI snapshot', () => {
    const title = create(<Title />);
    expect(title.toJSON()).toMatchSnapshot();
  });
});
