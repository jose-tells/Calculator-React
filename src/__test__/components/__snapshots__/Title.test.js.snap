// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> Title component UI snapshot 1`] = `
<header
  className="header"
>
  <h1
    className="header__title"
  >
    Hi!
    <br />
    What do you want to
    <br />
    calculate?
  </h1>
</header>
`;
